testing/features/step_definitions/navigation_steps.rb

```diff
@@ -0,0 +1,75 @@
+# frozen_string_literal: true
+
+Given("the navigation is on the page") do
+  @component = Navigation::Default.new.tap(&:load)
+end
+
+Given("the navigation is on the page with header links") do 
+  @component = Navigation::WithHeader.new.tap(&:load)
+end
+
+Given("the navigation does not fit on the screen") do 
+  page.driver.browser.manage.window.resize_to(320, 568)
+end
+
+Given("the dropdown menu is already open") do
+  page.driver.browser.manage.window.resize_to(320, 568)
+  click_button("More")
+end
+  
+Then("the dropdown toggle is not present") do 
+  expect(@component).to have_no_button("More")
+end
+
+Then("the dropdown toggle is present") do 
+  expect(@component).to have_button("More")
+end
+
+Then("the dropdown menu is hidden") do
+  expect(@component).to have_no_link("More from us")
+end
+
+Then("the dropdown menu is present") do 
+  expect(@component).to have_link("More from us")
+end
+
+Then("the Close button is present") do
+  expect(@component).to have_button("Close")
+end
+
+Then("the More button is present") do
+  expect(@component).to have_button("More")
+end
+
+Then("the header links should be included in the dropdown menu") do
+  expect(@component).to have_button("Sign out")
```

Comment:
For the signature here you can chain things. So you can do something like

`expect(@component.dropdown_menu).to have_header_links`